src/Control/HTTPRequest.php

```diff
@@ -564,7 +564,23 @@ public function match($pattern, $shiftOnSuccess = false)
 
                 /** @skipUpgrade */
                 $key = "Controller";
-                $arguments[$varName] = isset($this->dirParts[$i]) ? $this->dirParts[$i] : null;
+                if ($varName === '*') {
+                    array_pop($patternParts);
+                    $shiftCount = sizeof($patternParts);
+                    $patternParts = array_merge($patternParts, array_slice($this->dirParts, $i));
+                    break;
+                } elseif ($varName === '@') {
+                    array_pop($patternParts);
+                    $shiftCount = sizeof($patternParts);
+                    $remaining = count($this->dirParts) - $i;
+                    for ($j = 1; $j <= $remaining; $j++) {
+                        $arguments["$${j}"] = $this->dirParts[$j + $i - 1];
+                    }
+                    $patternParts = array_merge($patternParts, array_keys($arguments));
+                    break;
+                } else {
+                    $arguments[$varName] = isset($this->dirParts[$i]) ? $this->dirParts[$i] : null;
+                }
```

Comment:
How would you feel about putting this into a private method?

Reply:
I think without reworking most of the method it would be a bit of a pain because the logic has to mess with local variables in the loop to get this working. It is a bit hack-y, but maybe you could say that about all this pattern matching logic ;)